src/schemas/types/schema.ts

```diff
@@ -55,6 +63,10 @@ export type SchemaProperty = {
   maxItems?: number,
 }
 
+export function isSchemaProperty(value: SchemaProperty | SchemaProperty[] | undefined): value is SchemaProperty {
+  return isDefined(value) && !Array.isArray(value)
+}
```

Comment:
Do we need to introduce a way to convert from `SchemaProperty[]` to `SchemaProperty`? We'll need to maintain backwards compatibility with v1 for older versions of core.

Reply:
It crossed my mind to map `items: SchemaProperty[]` to `prefixItems` which is the correct way to do that. But for now I think what we have works fine.

Reply:
ok we can revisit as needed!